components/codec/src/columnar/mod.rs

```diff
@@ -680,6 +683,72 @@ mod tests {
         check_encode_end_decode(10, datums, DatumKind::Int64);
     }
 
+    #[test]
```

Comment:
Maybe we should add some cases for the bad case?
Such as zero columns or zero rows?